### PR TITLE
add all to StorageImpl

### PR DESCRIPTION
**Pull request by vlykhvar**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -4,12 +4,40 @@
...
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[STORAGE_CAPACITY];
+        this.values = (V[]) new Object[STORAGE_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < STORAGE_CAPACITY; i++) {
+            if (keys[i] == key) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:19]: is it the right way to compare two objects?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -4,12 +4,40 @@
...
+                values[i] = value;
+                break;
+            } else if (keys[i] == null && values[i] == null) {
+                keys[i] = key;
+                values[i] = value;
+                break;
+            }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:26]: we can unite these two conditions, right?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -4,12 +4,40 @@
...
     @Override
     public V get(K key) {
+        for (int i = 0; i < STORAGE_CAPACITY; i++) {
+            if (keys[i] != null && key == null) {
+                continue;
+            } else if (keys[i] == null && key == null) {
+                return values[i];
+            } else if (key.equals(keys[i])) {
+                return values[i];
+            }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:39]: can we write all these checks in a single `if`?
remember how we compared two objects in previous tasks (we compared fields there)

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -4,12 +4,41 @@
...
+                values[i] = value;
+                break;
+            } else if (keys[i] != null && keys[i].equals(key)) {
+                values[i] = value;
+                break;
+            } else if (keys[i] == null && values[i] == null) {
+                keys[i] = key;
+                values[i] = value;
+                break;
+            }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:29]: I mean the following:

*start of method*

for loop {
    if (key and value in arrays are both null OR input key and key from array are equal (both null or equal)) {
        replace key and value;
    }
}

*end of method*

what do you think?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -4,12 +4,41 @@
...
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < STORAGE_CAPACITY; i++) {
+            if (key != null && key.equals(keys[i])) {
+                return values[i];
+            } else if (key == null && keys[i] == null) {
+                return values[i];
+            }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:40]: let's see your previous solution (pair) and write everything in one `if` here

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]
great job!

**Comment by vlykhvar** [comment]
> great job!

thks, i almost think to push pc down from balcony

**Pull request closed without merging**